lua/mediaplayer/services/youtube/init.lua

```diff
@@ -134,18 +135,59 @@ function SERVICE:GetMetadata( callback )
 		callback(self._metadata)
 
 	else
-
 		local videoId = self:GetYouTubeVideoId()
-		local apiurl = MetadataUrl:format( videoId, APIKey )
+		local videoUrl = "https://www.youtube.com/watch?v="..videoId
 
-		self:Fetch( apiurl,
+		self:Fetch( videoUrl,
 			function( body, length, headers, code )
-				OnReceiveMetadata( self, callback, body )
+				local metadata = self:ParseYTMetaDataFromHTML(body, videoId)
+				
+				--html couldn't be parsed
+				if (!metadata.title || !metadata.duration) then
+					callback(false, "Failed to parse HTML Page for metadata")
+					return
+				end
+
+				self:SetMetadata(metadata, true)
+
+				if self:IsTimed() then
+					MediaPlayer.Metadata:Save(self)
+				end
+			
+				callback(self._metadata)
 			end,
 			function( code )
 				callback(false, "Failed to load YouTube ["..tostring(code).."]")
 			end
 		)
+	end
+end
+
+---
+-- Function to parse video metadata straight from the html instead of using the API
+--
+function SERVICE:ParseYTMetaDataFromHTML( html, videoId )
+	--Lua search patterns to find Title and Duration from the html mess
+	local titlepat = "\\\"title\\\":\\\".-\\\""
+	local durationpat = "\\\"lengthSeconds\\\":\\\".-\\\""
```

Comment:
There are some others methods of getting this information which would be less likely to break.

The first requirement would be to fetch the HTML using a different [User-Agent header](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/User-Agent). Using the [desktop Googlebot value](https://support.google.com/webmasters/answer/1061943?hl=en) should result in getting the pre-rendered HTML (more structured metadata, usually used for search engines).

The title can use `og:title` value (via [Open Graph Tags](https://ogp.me/)) or fallback to just the `<title>` tag.

The duration should be available in [Microdata schema markup](https://schema.org/docs/gs.html). I have [logic for that in another project I made](https://github.com/samuelmaddock/metastream/blob/master/packages/metastream-app/src/media/middleware/youtube.ts#L39) which also handles logic for livestreams.

Reply:
Thanks for the advice! I updated the function to get the info from those sources instead. Title and Thumbnail have fallback values now, and it sets the duration to 0 if it is a live broadcast.